osu.Game.Rulesets.Osu/Objects/OsuHitObject.cs

```diff
@@ -24,6 +24,9 @@ public abstract class OsuHitObject : HitObject, IHasCombo, IHasPosition
         public float X => Position.X;
         public float Y => Position.Y;
 
+        public float TimePreempt;
+        public float TimeFadein;

```

Comment:
Can we do something like `ApplyDifficulty` (or the usual `ApplyDefaults`) after creating the `DrawableHitObject` in the RulesetContainer? Then we have the values directly in the drawable and don't need to have these in the `HitObject`.

Reply:
I briefly discussed this with @peppy , I'm not too sure about it, we need to decide what should be a part of the HitObject and what should be a part of DrawableHitObject if we go down that route.

I wouldn't be against it, at least.

Reply:
FollowPointRenderer will be happy having them in HitObject as no static access is needed anymore.
But are they needed in both DrawableHitObject and HitObject simultaneously then?